pkg/controller/core/core.go

```diff
@@ -78,19 +65,13 @@ func SetupControllers(mgr ctrl.Manager, qManager *queue.Manager, cc *cache.Cache
 	if err := cqRec.SetupWithManager(mgr, cfg); err != nil {
 		return "ClusterQueue", err
 	}
-	ctrlOpts := ControllerOptions{
-		requeuingBaseDelaySeconds: defaultRequeuingBaseDelaySeconds,
-	}
-	for _, opt := range controllerOpts {
-		opt(&ctrlOpts)
-	}
 
 	if err := NewWorkloadReconciler(mgr.GetClient(), qManager, cc,
 		mgr.GetEventRecorderFor(constants.WorkloadControllerName),
 		WithWorkloadUpdateWatchers(qRec, cqRec),
 		WithPodsReadyTimeout(podsReadyTimeout(cfg)),
```

Comment:
Maybe it would be worth something like:

```
WithPodsReadyTimeout(cfg.WaitForPodsReady)
```

So we don't need to keep adding options for every single field.

Reply:
+1

Reply:
Good idea, I just pushed a commit along the lines, but I use:
`WithWaitForPodsReady(waitForPodsReady(cfg.WaitForPodsReady)),`
because the controller needs the entire config. Also I use the `waitForPodsReady` function to convert the underlying config into a structure easier to consume. In particular, I don't need the `enable` field.